lookout/core/metrics.py

```diff
@@ -1,12 +1,173 @@
 
-def submit_event(key: str, value: Union[int, float, bool]):
     """
-    Register another event by a key and with a numeric value.
+
+    def __init__(self):
+        """Initialize a new instance of PreciseFloat."""
+        self._value = 0.0
+        self._aux = 0
+        self._lock = Lock()
+
+    def __iadd__(self, amount):
+        """Increase the value by the given amount. Thread safe. No precision loss."""
+        with self._lock:
+            val = self._value
+            y = amount - self._aux
+            t = val + y
+            self._aux = (t - val) - y
+            self._value = t
+        return self
+
+    def set(self, value):
+        """Set the value to the given amount. Thread safe."""
+        with self._lock:
+            self._aux = 0
+            self._value = value
+
+    def get(self):
+        """Read the current value. Thread safe."""
+        with self._lock:
+            return self._value
+
+
+class ConfidentCounter(MetricWrapperBase):
+    """Prometheus Counter-like class.
+
+    It enables calculation of the rolling mean, and standard deviation.
+      This class internally reports the number of events, the sum of event values,
+      and the sum of squares of event values to Prometheus.
+    """
+
+    _type = "counter"
+
+    def __iadd__(self, amount):
+        """Add a delta value to the counter."""
+        self._count += 1
+        self._sum += amount
+        self._sum_of_squares += amount ** 2
+        return self
+
+    def _metric_init(self):
+        self._count = PreciseFloat()
+        self._sum = PreciseFloat()
+        self._sum_of_squares = PreciseFloat()
+
+    def _child_samples(self):
+        return (
+            ("_count", {}, self._count.get()),
+            ("_sum", {}, self._sum.get()),
+            ("_sum_of_squares", {}, self._sum_of_squares.get()),
+        )
+
+
+def start_prometheus(host: str, port: int):
+    """Start the prometheus HTTP Server on the specified address.
+
+    The stored metrics will be accessible at http://addr:port.
+
+    :param port: Target port to run the PrometheusServer.
+    :param host: IP address of the PrometheusServer. It defaults to localhost.
+    :return: None
+    """
+    global _prometheus_server
+    if _prometheus_server is None:
+        _prometheus_server = PrometheusServer(port=port, host=host)
+
+
+def submit_event(key: str, value: Union[int, float, bool], description: str = ""):
+    """Register an event by a key and with a numeric value.
+
+    If the key does not exist, it creates a new Prometheus Metric.
 
     :param key: Identifier of the event.
-    :param value: Value of the event.
+    :param value: Value of the event. It will convert cast variables to int.
+    :param description: Additional description of the event. Only used when creating a new event.
     :return: None
     """
-    pass
+    start_prometheus(host=PROMETHEUS_HOST, port=PROMETHEUS_PORT)
```

Comment:
```suggestion
    if _prometheus_server is None:
        _prometheus_server = PrometheusServer(host=PROMETHEUS_HOST, port=PROMETHEUS_PORT)
```

and remove `start_prometheus` function